app/logic/invitation_texts.rb

```diff
@@ -0,0 +1,240 @@
+      sms_pool += default_and_streak_conditions(protocol_completion, curidx) if sms_pool.empty?
+
+      sms_pool.sample
+    end
+
+    def mentor_message(_protocol, _protocol_completion)
+      # curidx = current_index(protocol_completion)
+      sms_pool = []
+
+      sms_pool.sample
+    end
+
+    def nth_response_pool(_curidx)
+      raise 'method nth_response_pool not implemented by subclass!'
+    end
+
+    def rewards_threshold_pool(_threshold)
+      raise 'method rewards_threshold_pool not implemented by subclass!'
+    end
+
+    def default_pool
+      raise 'method default_pool not implemented by subclass!'
+    end
+
+    def about_to_be_on_streak_pool
+      raise 'method about_to_be_on_streak_pool not implemented by subclass!'
+    end
+
+    def on_streak_pool
+      raise 'method on_streak_pool not implemented by subclass!'
+    end
+
+    def first_responses_missed_pool
+      raise 'method first_responses_missed_pool not implemented by subclass!'
+    end
+
+    def missed_last_pool
+      raise 'method missed_last_pool not implemented by subclass!'
+    end
+
+    def missed_more_than_one_pool
+      raise 'method missed_more_than_one_pool not implemented by subclass!'
+    end
+
+    def missed_everything_pool
+      raise 'method missed_everything_pool not implemented by subclass!'
+    end
+
+    def rejoined_after_missing_one_pool
+      raise 'method rejoined_after_missing_one_pool not implemented by subclass!'
+    end
+
+    def rejoined_after_missing_multiple_pool
+      raise 'method rejoined_after_missing_multiple_pool not implemented by subclass!'
+    end
+
+    def missed_after_streak_pool
+      raise 'method missed_after_streak_pool not implemented by subclass!'
+    end
+
+    private
+
+    def default_and_streak_conditions(protocol_completion, curidx)
+      sms_pool = []
+
+      # Streak about to be 3
+      sms_pool += about_to_be_on_streak_pool if protocol_completion[curidx][:streak] == STREAK_SIZE
+
+      # On bonus streak (== on streak > 3)
+      sms_pool += on_streak_pool if protocol_completion[curidx][:streak] > STREAK_SIZE && sms_pool.empty?
+
+      # Default messages
+      sms_pool += default_pool if sms_pool.empty?
+
+      sms_pool
+    end
+
+    def threshold_conditions(protocol, protocol_completion, curidx)
+      current_protocol_completion = truncated_protocol_completion(protocol_completion, curidx)
+      rewards_before = protocol.calculate_reward(current_protocol_completion, false)
+      rewards_after = protocol.calculate_reward(current_protocol_completion, true)
+
+      sms_pool = []
+      8.times do |idx|
+        threshold = (idx + 1) * 10 * 100
+        if rewards_before < threshold && rewards_after >= threshold
+          sms_pool += rewards_threshold_pool(threshold)
+        end
+      end
+      sms_pool
+    end
+
+    def special_conditions(protocol_completion, curidx)
+      sms_pool = []
```

Comment:
ja ik zou denk ik voor al die functies gewoon gelijk `return`en.

Reply:
Het is niet hetzelfde, zie comment boven.